pkg/codegen/dotnet/gen_program.go

```diff
@@ -1135,6 +1135,22 @@ func (g *generator) genResourceOptions(opts *pcl.ResourceOptions, resourceOption
 			} else {
 				g.Fgenf(&result, "\n%s%s = %v,", g.Indent, name, g.lowerExpression(value, value.Type()))
 			}
+		} else if name == "DependsOn" {
+			// depends on need to be special cased
+			// because new [] { resourceA, resourceB } cannot be implicitly casted to InputList<Resource>
```

Comment:
Might be worth a comment here why we're preferring the initialiser expression to `new Resource[] { A, B }`.

Reply:
I don't think it matters which syntax we go with, both implicitly convert to `InputList<Resource>` which is the type of the `DependsOn` field